src/tracing/metric_event.h

```diff
@@ -0,0 +1,128 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <matter/tracing/build_config.h>
+#include <tracing/metric_macros.h>
+#include <lib/core/CHIPError.h>
+#include <system/SystemClock.h>
+#include <tracing/metric_keys.h>
+
+namespace chip {
+namespace Tracing {
+
+/**
+ * Define a metric that can be logged. A metric consists of a key-value pair. The value is
+ * currently limited to simple scalar values. The value is interpreted based on the key type.
+ *
+ * Additionally a metric is tagged as either an instant type or marked with a begin and end
+ * for the event. When the latter is used, a duration can be associated between the two events.
+ */
+struct MetricEvent
+{
+    // Metric tag type
+    enum class Tag
+    {
+        Begin,  // Implies tracking a duration
```

Comment:
use `kUpperCamelCase` for constant naming

Reply:
Done